app/org/apache/spark/deploy/history/SparkFSFetcher.scala

```diff
@@ -147,7 +147,12 @@ class SparkFSFetcher(fetcherConfData: FetcherConfigurationData) extends Elephant
               null
             }
           } else {
-            val logFilePath = new Path(logPath + "_1.snappy")
+            val logFilePath =
+              getNoneSnappy(logPath) match {
```

Comment:
I think the switch case is making the code look complicated.

Why not make a call to a method, say getValidLogFilePath(logPath) which returns the correct path after checking all the types.